File: jcu-zoo/src/database/cage/CageDAO.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package database.cage;

import com.mysql.jdbc.Connection;

import com.mysql.jdbc.PreparedStatement;
import database.MySQLConnection;
import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.logging.Level;
import java.util.logging.Logger;
import model.Cage;
import model.Gate;

/**
 *
 * @author Panda
 */
public class CageDAO implements ICageDAO{
    String query = null;
    Connection conn = null;
    ResultSet resultSet = null;
    PreparedStatement preparedStatement = null;
    private static CageDAO cageDAO = null;
    private CageDAO(){}
    public static CageDAO getInstanceOf(){
        if (cageDAO == null){
            cageDAO = new CageDAO();
        }
        return cageDAO;
    }
    private void connect(){
        try {
            conn = MySQLConnection.connectDB();
        } catch (Exception ex) {
            Logger.getLogger(CageDAO.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
    
    private void cleanUp(){      
        this.query = null;
        closeResultSet();
        closePreparedStatement();
        closeConnection();
    }
    private void closeResultSet(){
        try {
                if (resultSet != null){
                    resultSet.close();
                    resultSet = null;
                }
        } catch (Exception ex){
            Logger.getLogger(CageDAO.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
    private void closePreparedStatement(){
        try {
                if (preparedStatement != null){
                    preparedStatement.close();
                    preparedStatement = null;
                }
        } catch (Exception ex){
            Logger.getLogger(CageDAO.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
    private void closeConnection(){
        try {
            if (conn != null){
                conn.close();
            }
        } catch (Exception ex) {
            Logger.getLogger(CageDAO.class.getName()).log(Level.SEVERE, null, ex);
        }        
    }
    /* @Override
    public ArrayList<Cage> getCages() {
        //throw new UnsupportedOperationException("Not supported yet.");
        ArrayList<Cage> cages = new ArrayList<Cage>();
         try {
            connect();
            query = "SELECT * FROM CAGE";
            preparedStatement = (PreparedStatement) conn.prepareStatement(query);
            resultSet = preparedStatement.executeQuery();
            while (resultSet.next()){
                while (resultSet.next()){
                    cages.add(new Cage(resultSet.getInt("CAGE_ID"),
                            (String) resultSet.getString("CAGE_NAME"),
                            (Float) resultSet.getFloat("LATITUDE"),
                            (Float) resultSet.getFloat("LONGITUDE"),
                            (String) resultSet.getString("CAGE_TYPE"),
                            (Boolean) resultSet.getBoolean("HAS_HUMAN"),
                            (Boolean) resultSet.getBoolean("HAS_ANIMAL"),
                            (String) resultSet.getString("EXHIBIT_NAME"),
                            (String) resultSet.getString("EXHIBIT_DESCRIPTION")));
                }
            }
        } catch (Exception ex){
            Logger.getLogger(CageDAO.class.getName()).log(Level.SEVERE, null, ex);
        } finally {
            cleanUp();
        }        
         return cages;
    }*/

    @Override
    public ArrayList<Gate> getGates(Cage cage) {
        ArrayList<Gate> gates = new ArrayList<Gate>();
        try {
            connect();
            query = "SELECT GATE.GATE_ID, GATE.IS_CLOSED, GATE.IS_LOCKED FROM GATE,CAGE_GATES WHERE CAGE_GATES.CAGE_ID=? AND GATE.GATE_ID=CAGE_GATES.GATE_ID";
            preparedStatement = (PreparedStatement) conn.prepareStatement(query);
            preparedStatement.setInt(1, cage.getCageId());
            resultSet = preparedStatement.executeQuery();                        
            while (resultSet.next()){
                gates.add(new Gate(resultSet.getInt("GATE_ID"),resultSet.getBoolean("IS_CLOSED"),resultSet.getBoolean("IS_LOCKED")));
            }
                    
        } catch (Exception ex){
            Logger.getLogger(CageDAO.class.getName()).log(Level.SEVERE, null, ex);
        } finally {
            cleanUp();
        }
        return gates;
    }

    @Override
    public void updateGate(Gate gate) {
        try {
            query = "UPDATE GATE SET IS_CLOSED=?, IS_LOCKED=? WHERE GATE_ID=?";
            preparedStatement = (PreparedStatement) conn.prepareStatement(query);
            preparedStatement.setBoolean(1, gate.isClosed());
            preparedStatement.setBoolean(2, gate.isLocked());
            preparedStatement.setInt(3,gate.getGateId());
            preparedStatement.executeUpdate();
            
        } catch (Exception ex){
            Logger.getLogger(CageDAO.class.getName()).log(Level.SEVERE, null, ex);
        } finally {
            cleanUp();
        }
    }
	@Override
	public Gate getGate(int gateId) {
		Gate gate = null;
		try {
            query = "SELECT * FROM GATE WHERE GATE_ID=?";
            preparedStatement = (PreparedStatement) conn.prepareStatement(query);
            preparedStatement.setInt(1, gateId);
            resultSet = preparedStatement.executeQuery();
            while (resultSet.next()){
            	gate = new Gate(resultSet.getInt("GATE_ID"),resultSet.getBoolean("IS_CLOSED"),resultSet.getBoolean("IS_LOCKED"));
            }
		} catch (Exception ex){
			Logger.getLogger(CageDAO.class.getName()).log(Level.SEVERE, null, ex);
		} finally {
			cleanUp();
		}
		return gate;
	}
    
}

File: jcu-zoo-php/class/CageDAO.php
<?php

/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of CageDAO
 *
 * @author Panda
 */
include_once('class/MySQLConnection.php'); 
include_once('class/Cage.php');
class CageDAO {
    private static $conn;

    public function __construct(){
        $dbh = new MySQLConnection();
        if (!isset(self::$conn)){
            self::$conn = $dbh->getConnection();        
        }
    }
    function getGates($cage){
        $gates;
        $query = "SELECT GATE.GATE_ID, GATE.IS_CLOSED, GATE.IS_LOCKED FROM GATE,CAGE_GATES WHERE CAGE_GATES.CAGE_ID=:cageId AND GATE.GATE_ID=CAGE_GATES.GATE_ID";
        $statement = self::$conn->prepare($query);
        $statement->bindParam(':cageId',$cage->getCageId(),SQL_INTEGER);
        $statement->execute();
        foreach ($statement->fetchAll() as $gate){
            $gates[] = new Gate($gate["GATE_ID"], $gate["IS_CLOSED"], $gate["IS_LOCKED"]);
        }
        return $gates;
    }
    function updateGate($gate){
        // update gate
        $query = "UPDATE GATE SET IS_CLOSED=:isClosed, IS_LOCKED=:isLocked WHERE GATE_ID=:gateId";
        $statement = self::$conn->prepare($query);
        $statement->bindParam(':isClosed',$gate->isClosed(),SQL_TINYINT);
        $statement->bindParam(':isLocked',$gate->isLocked(),SQL_TINYINT);
        $statement->bindParam(':gateId',$gate->getGateId(),SQL_INTEGER);
        $statement->execute();
    }
    function getGate($gateId){
        //get gate
        $query = "SELECT * FROM GATE WHERE GATE_ID=:gateId";
        $statement = self::$conn->prepare($query);
        $statement->bindParam(':gateId',$gateId,SQL_INTEGER);
        $statement->execute();
        $gate = $statement->fetch();
        return new Gate($gate["GATE_ID"], $gate["IS_CLOSED"], $gate["IS_LOCKED"]);
        
    }
        function getCages(){
            $cages;
            $query = "SELECT * FROM CAGE";
            $statement = self::$conn->prepare($query);
            $statement->execute();
            $result = $statement->fetchAll();
             $statement = null;
             foreach ($result as $cage){
                 $cages[] = new Cage($cage['CAGE_ID'],$cage['CAGE_NAME'],$cage['EXHIBIT_NAME'],$cage['EXHIBIT_DESCRIPTION'],$cage['LATITUDE'],$cage['LONGITUDE'],$cage['HAS_ANIMAL'],$cage['HAS_HUMAN'],$cage["CAGE_TYPE"]);
             }
             return $cages;
    }
    function getCage($cageId) {
        $cage;
        $query = "SELECT * FROM CAGE WHERE CAGE_ID=:cageId";
        $statement = self::$conn->prepare($query);
        $statement->bindParam(':cageId',$cageId,SQL_INTEGER);
        $statement->execute();
        $cage = $statement->fetch();
        $statement = null;
        return new Cage($cage['CAGE_ID'],$cage['CAGE_NAME'],$cage['EXHIBIT_NAME'],$cage['EXHIBIT_DESCRIPTION'],$cage['LATITUDE'],$cage['LONGITUDE'],$cage['HAS_ANIMAL'],$cage['HAS_HUMAN'],$cage["CAGE_TYPE"]);
    }
        function getNumOfCages(){
        $query = "SELECT COUNT(*) FROM CAGE";
        $statement = self::$conn->prepare($query);
        $statement->execute();
        $result = $statement->fetch();
        $statement = null;

        return $result[0];
    }
        function updateCage(Cage $cage){
        $query = "UPDATE CAGE SET CAGE_NAME=:cageName, LATITUDE=:latitude, LONGITUDE=:longitude, CAGE_TYPE=:cageType, HAS_HUMAN=:hasHuman, HAS_ANIMAL=:hasAnimal, EXHIBIT_NAME=:exhibitName, EXHIBIT_DESCRIPTION=:exhibitDesc WHERE CAGE_ID=:cageId";
        $statement = self::$conn->prepare($query);
        $statement->bindParam(':cageName',$cage->getCageName());
        $statement->bindParam(':latitude',$cage->getLatitude());
        $statement->bindParam(':longitude',$cage->getLongitude());
        $statement->bindParam(':cageType',$cage->getCageType());
        $statement->bindParam(':hasHuman',$cage->hasHuman());
        $statement->bindParam(':hasAnimal',$cage->hasAnimal());
        $statement->bindParam(':exhibitName',$cage->getExhibitName());
        $statement->bindParam(':exhibitDesc',$cage->getExhibitDesc());
        $statement->bindParam(':cageId',$cage->getCageId());
        $statement->execute();
        $statement = null;
    }
}

?>
